sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/SparkSqlLexer.g

```diff
@@ -328,6 +328,8 @@ KW_WEEK: 'WEEK'|'WEEKS';
 KW_MILLISECOND: 'MILLISECOND'|'MILLISECONDS';
 KW_MICROSECOND: 'MICROSECOND'|'MICROSECONDS';
 
+KW_NATURAL: 'NATURAL';
```

Comment:
I think it should be reserved.
See:
MySQL: https://dev.mysql.com/doc/refman/5.5/en/keywords.html
PostgreSQL: http://www.postgresql.org/docs/7.3/static/sql-keywords-appendix.html

and it's also reserved in SQL92 and SQL99(as declared in the doc of PostgreSQL).